go/arrow/array/compare.go

```diff
@@ -450,6 +451,13 @@ func WithAbsTolerance(atol float64) EqualOption {
 	}
 }
 
+// WithUnorderedMapKeys configures the comparison functions so that Map with different entries order are considered equal.
+func WithUnorderedMapKeys(v bool) EqualOption {
+	return func(o *equalOption) {
+		o.unorderedMapKeys = v
+	}
+}
```

Comment:
Nope (as it's stated in the `KeysSorted` doc -- it's only for the user private purposes, it's not used in the comparison (and this would definitely become a breaking change if we use that).